### PR TITLE
WIP adds cypress, modifies files, adds pages

### PR DESCRIPTION
**Pull request by Josephcabs**
- **feat: adds tests, new endpoints, new page changes. -JosephCaballero**
- **modifies cypress, makes first test, moves clerk to appropriate space, modifies list items, adds appropriate stuff for cypress and later use of code coverage. handles merge conflicts and improves upon them. -JosephCaballero & Angie**

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { defineConfig } from 'cypress';
+import codeCoverageTask from '@cypress/code-coverage/task';
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
+      codeCoverageTask(on, config);
+      return config;
+    },
+    baseUrl: 'http://localhost:3000',
```
Comment (XavierCruz5106) [inline cypress.config.ts:24]: will have to change for deployment. As of rn its good however

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// declare global {
+//   namespace Cypress {
+//     interface Chainable {
+//       login(email: string, password: string): Chainable<void>
+//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
+//     }
+//   }
+// }
```
Comment (XavierCruz5106) [inline cypress/support/commands.ts:37]: If this is going to be unused in the final product, consider removing

File: src/app/api/history/route.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import getDatabaseConnection from "../scripts/db";
+import { NextRequest, NextResponse } from "next/server";
+import { CartItem } from "@/types/cartItem";
+
+const db = getDatabaseConnection();
+
+export async function GET(){
+    const all = db.prepare(`SELECT * FROM history`).all();
+    if(all == null){
+        return NextResponse.json("nah")
```
Comment (XavierCruz5106) [inline src/app/api/history/route.ts:10]: lol

**Review by XavierCruz5106 — APPROVED** [review]
LGTM

**Review by marleomac3 — APPROVED** [review]

**Review by XavierCruz5106 — APPROVED** [review]
Holy Hell brother. Imma trust it works. YOLO

**Pull request merged**